pom.xml

```diff
@@ -27,24 +28,40 @@
       <dependency>
         <groupId>commons-logging</groupId>
         <artifactId>commons-logging</artifactId>
-        <version>1.1.1</version>
+        <version>1.2</version>
       </dependency>
-    </dependencies>
+      <dependency>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-api</artifactId>
+          <version>1.7.36</version>
+      </dependency>
+      </dependencies>
   </dependencyManagement>
+  
   <dependencies>
     <dependency>
       <groupId>com.esri.geoevent.sdk</groupId>
       <artifactId>geoevent-sdk</artifactId>
       <version>${project.version}</version>
       <scope>provided</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>log4j</groupId>
+          <artifactId>log4j</artifactId>
+        </exclusion>
+      </exclusions> 
+    </dependency>
+    <dependency>
+        <groupId>ch.qos.reload4j</groupId>
+        <artifactId>reload4j</artifactId>
+        <version>1.2.19</version>
```

Comment:
Can you try replacing that with a subset of (and scope `provided` when you can):
```
org.slf4j:slf4j-api:1.7.32 --> you can try v1.7.35 as well
org.slf4j:slf4j-log4j12:1.7.32 --> only if needed.
org.apache.logging.log4j:log4j-slf4j-impl:2.17.1 -> only if needed
org.apache.logging.log4j:log4j-core:2.17.1
org.apache.logging.log4j:log4j-api:2.17.1
org.apache.logging.log4j:log4j-1.2-api:2.17.1 --> this is the log4j to log4j2 API bridge
```

Reply:
Keep in mind that this component supports GeoEvent Server 10.4.1 and above, and is built against the SDK as published on Maven Central, independent of what patches are installed on the development machine.

However, if you are concerned it would cause a problem to have those files embedded, I'm OK with changing the dependency to provided. Then, at run time it relies on the underlying GeoEvent Server to be patched and/or stripped of the problematic classes. I just wasn't sure what would happen if it was actually the original log4j:log4j that was provided. 
@hanoch do you think that would be better?

For slf4j, I understood that it was at v1.7.35 where they initially switched to reload4j; however, as v1.7.36 was released a month later, and uses the latest reload4j 1.2.19, I thought probably best to use that one.

Reply:
Actually, perhaps best to mark reload4j as both provided and also optional, and also specify that slf4j is provided, even though a version is specified. Then neither module is included in the JAR, and it is up to GeoEvent Server to provide these classes at runtime, as before. (I'm not sure if the 'optional' is strictly needed when scope is 'provided' anyway, but thought it might be safer / clearer, since the classes might be supplied by log4j rather than reload4j in an older version.)